ql/src/java/org/apache/hadoop/hive/ql/stats/StatsUpdaterThread.java

```diff
@@ -220,6 +221,16 @@ private void stopWorkers() {
     String skipParam = table.getParameters().get(SKIP_STATS_AUTOUPDATE_PROPERTY);
     if ("true".equalsIgnoreCase(skipParam)) return null;
 
+    // If the table is being replicated into,
+    // 1. the stats are also replicated from the source, so we don't need those to be calculated
+    //    on the target again
+    // 2. updating stats requires a writeId to be created. Hence writeIds on source and target
+    //    can get out of sync when stats are updated. That can cause consistency issues.
+    String replTrgtParam = table.getParameters().get(ReplConst.REPL_TARGET_PROPERTY);
+    if (replTrgtParam != null && !replTrgtParam.isEmpty()) {
+      return null;
```

Comment:
Added a debug instead. There's already a debug "Processing table", so adding an info didn't look symmetric. On a replicated database, this would produce a lot of INFOs since all the tables in that database will be target of replication.

Reply:
ok